rust/tests/optimize_test.rs

```diff
@@ -0,0 +1,506 @@
+    /// Validate Metrics when no files are optimized
+    async fn test_idempotent_metrics() -> Result<(), Box<dyn Error>> {
+        let context = setup_test(true).await?;
+        let mut dt = context.table;
+        let mut writer = RecordBatchWriter::for_table(&dt, HashMap::new())?;
+
+        write(
+            &mut writer,
+            &mut dt,
+            generate_random_batch(records_for_size(1_000_000), "2022-05-22")?,
+        )
+        .await?;
+
+        let version = dt.version;
+        let optimize = Optimize::default().target_size(10_000_000);
+        let metrics = optimize.execute(&mut dt).await?;
+
+        let mut expected_metric_details = MetricDetails::default();
+        expected_metric_details.min = 0;
+        expected_metric_details.max = 0;
+        expected_metric_details.avg = 0.0;
+        expected_metric_details.total_files = 0;
+        expected_metric_details.total_size = 0;
+
+        let mut expected = Metrics::default();
+        expected.num_files_added = 0;
+        expected.num_files_removed = 0;
+        expected.partitions_optimized = 0;
+        expected.num_batches = 0;
+        expected.total_considered_files = 1;
+        expected.total_files_skipped = 1;
+        expected.preserve_insertion_order = true;
+        expected.files_added = expected_metric_details.clone();
+        expected.files_removed = expected_metric_details.clone();
+
+        assert_eq!(expected, metrics);
+        assert_eq!(version, dt.version);
+        Ok(())
+    }
+
+    #[tokio::test]
+    /// Validate operation data and metadata was written
+    async fn test_commit_info() -> Result<(), Box<dyn Error>> {
+        let context = setup_test(true).await?;
+        let mut dt = context.table;
+        let mut writer = RecordBatchWriter::for_table(&dt, HashMap::new())?;
+
+        write(
+            &mut writer,
+            &mut dt,
+            tuples_to_batch(vec![(1, 2), (1, 3), (1, 4)], "2022-05-22")?,
+        )
+        .await?;
+
+        write(
+            &mut writer,
+            &mut dt,
+            tuples_to_batch(vec![(2, 2), (2, 3), (2, 4)], "2022-05-22")?,
+        )
+        .await?;
+
+        let version = dt.version;
+
+        let filter = vec![PartitionFilter::try_from(("date", "=", "2022-05-22"))?];
+
+        let optimize = Optimize::default().target_size(2_000_000).filter(&filter);
+        let metrics = optimize.execute(&mut dt).await?;
+
+        let commit_info = dt.history(None).await?;
+        let last_commit = &commit_info[commit_info.len() - 1];
+
+        let commit_metrics =
+            serde_json::from_value::<Metrics>(last_commit["operationMetrics"].clone())?;
+
+        assert_eq!(commit_metrics, metrics);
+        assert_eq!(last_commit["readVersion"], json!(version));
+        assert_eq!(
+            last_commit["operationParameters"]["targetSize"],
+            json!(2_000_000)
+        );
+        assert_eq!(last_commit["operationParameters"]["predicate"], Value::Null);
```

Comment:
Should predicate equal the filter used above?

Reply:
Yes it should. I added an additional TODO here. There's isn't a function that obtains the String representation of PartitionFilters. Should be fairly simple to implement.

Reply:
Okay. That seems sufficient for now.